src/OrleansCounterControl/OrleansCounterControl.csproj

```diff
@@ -45,8 +45,24 @@
     <WarningsAsErrors>4014</WarningsAsErrors>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```

Comment:
You keep asserting that "conclusion" @sergeybykov But as far as i can recall [or find using search] there has never been anything documented and/or agreed decision that changes & reverses the project development practices that are written down on this project's "Contributing" pages:
http://dotnet.github.io/orleans/Contributing

The thread below is the only canonical discussion i could find about assembly soft references. 
That thread even shows specific concrete examples where the VS / NuGet tools was demonstrably doing **wrong** things -- and that conclusion was not just mine but confirmed / agreed by others too. 
https://github.com/dotnet/orleans/pull/1115

Do you have a citation for the asserted "conclusion" ?